kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaOutputOperator.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.apex.malhar.kafka;
+
+import java.util.Properties;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Operator;
+
+public abstract class AbstractKafkaOutputOperator<K, V> implements Operator
+{
+  @SuppressWarnings("unused")
+  private static final Logger logger = LoggerFactory.getLogger(AbstractKafkaOutputOperator.class);
+  private transient Producer<K, V> producer;  // K is key partitioner, V is value type
+  @NotNull
+  private String topic = "topic1";
+  private String producerProperties = "";
+  private Properties configProperties = new Properties();
+  public Properties getConfigProperties()
+  {
+    return configProperties;
+  }
+
+  public void setConfigProperties(Properties configProperties)
+  {
+    this.configProperties.putAll(configProperties);
+  }
+
+  /**
+   * setup producer configuration.
+   * @return ProducerConfig
+   */
+  protected Properties createKafkaProducerConfig()
+  {
+    Properties prop = new Properties();
+    for (String propString : producerProperties.split(",")) {
+      if (!propString.contains("=")) {
+        continue;
+      }
+      String[] keyVal = StringUtils.trim(propString).split("=");
+      prop.put(StringUtils.trim(keyVal[0]), StringUtils.trim(keyVal[1]));
+    }
+
+    configProperties.putAll(prop);
+
+    return configProperties;
+  }
+
+  public Producer<K, V> getProducer()
```

Comment:
Any reason why this is made public? Its being accessed from subclass.. So can be made protected.